app/validation/validator.py

```diff
@@ -58,58 +73,40 @@ def _validate_json_against_schema(self, json_to_validate):
         except SchemaError as e:
             return '{}'.format(e)
 
-    def validate_calculated_ids_in_answers_to_calculate_exists(self, json_to_validate):
+    def validate_calculated_ids_in_answers_to_calculate_exists(self, question):
         # Validates that any answer ids within the 'answer_to_group'
         # list are existing answers within the question
 
         errors = []
 
-        for question in self._get_questions(json_to_validate):
-            if question['type'] == 'Calculated':
-                answer_ids = [answer['id'] for answer in question.get('answers')]
-                calculated_question = question.get('calculated')
-                for answer_id in calculated_question['answers_to_calculate']:
-                    if answer_id not in answer_ids:
-                        invalid_answer_id_error = 'Answer id - {} does not exist within this question - {}'\
-                            .format(answer_id, question['id'])
-                        errors.append(self._error_message(invalid_answer_id_error))
+        if question['type'] == 'Calculated':
+            answer_ids = [answer['id'] for answer in question.get('answers')]
+            calculated_question = question.get('calculated')
+            for answer_id in calculated_question['answers_to_calculate']:
+                if answer_id not in answer_ids:
+                    invalid_answer_id_error = 'Answer id - {} does not exist within this question - {}'\
+                        .format(answer_id, question['id'])
+                    errors.append(self._error_message(invalid_answer_id_error))
 
         return errors
 
-    def validate_schema_contains_valid_routing_rules(self, json_to_validate):
+    def validate_schema_contains_valid_routing_rules(self, groups, blocks, rule):
 
         errors = []
 
-        blocks = self._get_blocks(json_to_validate)
-        for block in blocks:
-            if 'routing_rules' in block and block['routing_rules']:
-                for rule in block['routing_rules']:
-                    if 'goto' in rule and 'id' in rule['goto'].keys():
-                        block_id = rule['goto']['id']
-                        if block_id == 'summary':
-                            continue
+        if 'goto' in rule and 'id' in rule['goto'].keys():
+            block_id = rule['goto']['id']
```

Comment:
'id' is never in a goto rule. It should be 'block'. See: [goto definition](https://github.com/ONSdigital/eq-schema-validator/blob/master/schemas/common_definitions.json#L252)